src/common/bustub_instance.cpp

```diff
@@ -0,0 +1,80 @@
+#include "common/bustub_instance.h"
+#include <fmt/format.h>
+#include "binder/binder.h"
+#include "catalog/table_generator.h"
+#include "recovery/log_manager.h"
+
+namespace bustub {
+
+auto BustubInstance::GetExecutionContext(Transaction *txn) -> std::unique_ptr<ExecutorContext> {
+  return std::make_unique<ExecutorContext>(txn, catalog_, buffer_pool_manager_, transaction_manager_, lock_manager_);
+}
+
+BustubInstance::BustubInstance(const std::string &db_file_name) {
+  enable_logging = false;
+
+  // storage related
+  disk_manager_ = new DiskManager(db_file_name);
+
+  // log related
+  log_manager_ = new LogManager(disk_manager_);
+
+  buffer_pool_manager_ = new BufferPoolManagerInstance(BUFFER_POOL_SIZE, disk_manager_, log_manager_);
+
+  // txn related
```

Comment:
```suggestion
  // Transaction (txn) related.
```

I normally wouldn't bother spelling txn out, but it is plausible that students will read this code.